DataFramework/Context/ProbabilityContext.Accessors.cs

```diff
@@ -13,17 +15,17 @@ public static class ProbabilityContextAccessors
 		/// </summary>
 		/// <param name="searchForPool"></param>
 		/// <returns></returns>
-		public static int? GetPoolId(ProbabilityContext context, List<PoolDie> searchForPool) =>
-			searchForPool.Select(die => context.PoolDice.Where(w => w.DieId == die.DieId && w.Quantity == die.Quantity && w.Pool.PoolDice.Count == searchForPool.Count())
-			.Select(s => s.PoolId)).Aggregate((result, next) => result.Intersect(next)).FirstOrDefault();
+		public static async Task<int?> GetPoolId(ProbabilityContext context, List<PoolDie> searchForPool) =>
+			await searchForPool.Select(die => context.PoolDice.Where(w => w.DieId == die.DieId && w.Quantity == die.Quantity && w.Pool.PoolDice.Count == searchForPool.Count())
+			.Select(s => s.PoolId)).Aggregate((result, next) => result.Intersect(next)).FirstOrDefaultAsync();
```

Comment:
Are you over-awaiting here?
```suggestion
		public static Task<int?> GetPoolId(ProbabilityContext context, List<PoolDie> searchForPool) =>
			searchForPool.Select(die => context.PoolDice.Where(w => w.DieId == die.DieId && w.Quantity == die.Quantity && w.Pool.PoolDice.Count == searchForPool.Count())
			.Select(s => s.PoolId)).Aggregate((result, next) => result.Intersect(next)).FirstOrDefaultAsync();
```

Reply:
You'd think but the compiler would complain about return types if I tried to pass the task down directly.  `FirstOrDefaultAsync` doesn't return a nullable integer.

Reply:
Good to know! Then yea, everything looks good.